src/main/java/org/broadinstitute/hellbender/utils/IntervalUtils.java

```diff
@@ -1109,4 +1113,262 @@ static public int endOfShard(int shardIndex, int shardSize) {
+     * @param locatables2 list of locatables
+     * @return Locatables from the unioned breakpoints of locatable1 and locatable2.  If both inputs are null, return an
+     *   empty list.  Please note that returned values are new copies.
+     */
+    static public <T extends Locatable> List<Locatable> combineBreakpoints(final List<T> locatables1, final List<T> locatables2) {
+        if ((locatables1 == null) && (locatables2 == null)) {
+            return Collections.emptyList();
+        }
+
+        validateNoOverlappingIntervals(locatables1);
+        validateNoOverlappingIntervals(locatables2);
+
+        if (CollectionUtils.isEmpty(locatables1)) {
+            return locatables2.stream().map(SimpleInterval::new).collect(Collectors.toList());
+        }
+        if (CollectionUtils.isEmpty(locatables2)) {
+            return locatables1.stream().map(SimpleInterval::new).collect(Collectors.toList());
+        }
+
+        final List<Locatable> masterList = new ArrayList<>();
+        masterList.addAll(locatables1);
+        masterList.addAll(locatables2);
+
+        final Set<String> contigs = masterList.stream()
+                .map(Locatable::getContig).collect(Collectors.toSet());
+
+        final Map<String, Set<Pair<Integer, IntervalBreakpointTypeEnum>>> contigToBreakpoints = contigs.stream()
+                .collect(Collectors.toMap(Function.identity(), l -> new HashSet<>()));
+
+        // Populate initialized maps with contigs to the break points.  Also, keep a mapping of contigs
+        //  to start breakpoints.
+        masterList.forEach(l -> {
+            contigToBreakpoints.get(l.getContig()).add(Pair.of(l.getStart(), IntervalBreakpointTypeEnum.START_BREAKPOINT));
+            contigToBreakpoints.get(l.getContig()).add(Pair.of(l.getEnd(), IntervalBreakpointTypeEnum.END_BREAKPOINT));
+        });
+
+        final List<Locatable> result = new ArrayList<>();
+
+        for (final String contig : contigs) {
+
+            // Sort the breakpoints for this contig.  Use the pair structure, since we need to differentiate between a
+            //  breakpoint that is a start and a breakpoint that is end, yet have the same position.  This is especially
+            //  important for single base intervals.
+            // TODO: Shorten the code here
+            final List<Pair<Integer, IntervalBreakpointTypeEnum>> breakpoints = new ArrayList<>(contigToBreakpoints.get(contig));
+            breakpoints.sort((p1, p2) -> {
+                final int firstComparison = p1.getLeft().compareTo(p2.getLeft());
+                if (firstComparison != 0) {
+                    return firstComparison;
+                } else {
+                    // We want start before end
+                    return p1.getRight().compareTo(p2.getRight());
+                }
+            });
+            int numCurrentStarts = 0;
+            int numCurrentEnds = 0;
+            for (int i = 0; i < (breakpoints.size() - 1); i++) {
```

Comment:
Just for the record, I have not gone through the logic here carefully.  This kind of code is pretty tricky (at least for me), and the less of it we have in the codebase the better, I think.  (I'm not even sure that my implementation in SegmentUtils was 100% bug free!)  I'm going to assume that the tests have covered all possible edge cases and that this code is completely correct for now...